examples/make_eeg_bids.py

```diff
@@ -0,0 +1,199 @@
+# MNE-Python:
+
+# Make a directory to save the data to
+home = os.path.expanduser('~')
+mne_dir = os.path.join(home, 'mne_data')
+if not os.path.exists(mne_dir):
+    os.makedirs(mne_dir)
+
+# Define which tasks we want to download.
+tasks = [2,  # This is 2 minutes of eyes closed rest
+         4,  # This is run #1 of imagining to close left or right fist
+         12]  # This is run #2 of imagining to close left or right fist
+
+# Download the data for subjects 1 and 2
+for subj_idx in [1, 2]:
+    eegbci.load_data(subject=subj_idx,
+                     runs=tasks,
+                     path=mne_dir,
+                     update_path=True)
+
+###############################################################################
+# Let's see whether the data has been downloaded using a quick visualization
+# of the directory tree.
+
+data_dir = os.path.join(mne_dir, 'MNE-eegbci-data')
+print_dir_tree(data_dir)
+
+###############################################################################
+# The data are in the `European Data Format <https://www.edfplus.info/>`_
+# '.edf', which is good for us because next to the BrainVision format, EDF is
+# one of the recommended file formats for EEG BIDS. However, apart from the
+# data format, we need to build a directory structure and supply meta data
+# files to properly *bidsify* this data.
+
+###############################################################################
+# Step 2: Formatting as BIDS
+# --------------------------
+#
+# Let's start by formatting a single subject. We are reading the data using
+# MNE-Python's io module and the `read_raw_edf` function.
+data_dir = os.path.join(data_dir, 'physiobank', 'database', 'eegmmidb')
+edf_path = os.path.join(data_dir, 'S001', 'S001R02.edf')
+raw = read_raw_edf(edf_path, preload=True)
+print(raw)
+
+###############################################################################
+# With this simple step we have everything to start a new BIDS directory using
+# our data. To do that, we can use the high level function `raw_to_bids`, which
+# is the core of MNE-BIDS. Generally, `raw_to_bids` tries to extract as much
+# meta data as possible from the raw data and then formats it in a BIDS
+# compatible way. `raw_to_bids` takes a bunch of inputs, most of which are
+# however optional. The required inputs are:
+#
+# * subject_id
+# * task
+# * raw_file
+# * output_path
+#
+# ... as you can see in the docstring:
+help(raw_to_bids)
+
+###############################################################################
+# We loaded 'S001R02.edf', which corresponds to subject 1 in the second task.
+# The second task was to rest with closed eyes.
+subject_id = '001'  # zero padding to account for >100 subjects in this dataset
+task = 'resteyesclosed'
+raw_file = raw
+output_path = os.path.join(home, 'mne_data', 'eegmmidb_bids')
+
+# Clean up existing directories for the sake of the example
+if os.path.exists(output_path):
+    sh.rmtree(output_path)
```

Comment:
why is this needed? Is it related to our `overwrite` discussion?

Reply:
yes exactly

Reply:
yes but we really need to show an example with incremental addition of data? I would delay it until we've ironed out the `overwrite` API. I don't want to give users ideas ... :)